src/main/java/dk/kb/datahandler/enrichment/DataEnricher.java

```diff
@@ -0,0 +1,96 @@
+package dk.kb.datahandler.enrichment;
+
+import dk.kb.datahandler.oai.OaiHarvestClient;
+import dk.kb.datahandler.util.PreservicaUtils;
+import dk.kb.storage.model.v1.DsRecordDto;
+import dk.kb.util.xml.XML;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.w3c.dom.*;
+import org.xml.sax.SAXException;
+
+import javax.xml.parsers.ParserConfigurationException;
+import javax.xml.transform.TransformerException;
+import java.io.IOException;
+import java.net.URISyntaxException;
+import java.util.List;
+
+public class DataEnricher {
+
+    private static final Logger log = LoggerFactory.getLogger(DataEnricher.class);
+
+
+    public static DsRecordDto apply(DsRecordDto record) {
+        Document recordData;
+        try {
+            recordData = XML.fromXML(record.getData(),true);
+            List<Fragment> fragments = FragmentsClient.getInstance().fetchMetadataFragments(getIoId(record));
+            if (!fragments.isEmpty()) {
+                log.info("No fragments found "+record.getId());
+            }
+            for (Fragment fragment : fragments) {
+                Document fragmentDoc = XML.fromXML(fragment.getMetadataFragment(),true);
+                fragmentDoc.getElementById("record");
+                addMetadataFragments(recordData, fragmentDoc);
+            }
+            record.setData(XML.domToString(recordData));
+            // TODO error handling
+        } catch (ParserConfigurationException e) {
+            throw new RuntimeException(e);
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        } catch (SAXException e) {
+            throw new RuntimeException(e);
+        } catch (URISyntaxException e) {
+            throw new RuntimeException(e);
+        } catch (TransformerException e) {
+            throw new RuntimeException(e);
+        }
+        //   String enrichedData = addMetadataFragments(recordData,fragments);
+    //    record.setData(enrichedData);
+        return record;
+    }
+
+    private static Document addMetadataFragments(Document record, Document fragments) {
+        // TODO check if fragment exists
+        NodeList nodeList = record.getElementsByTagName("XIP");
+        if (nodeList.getLength() > 0 ) {
+            Node xipNode = nodeList.item(0);
+            Element metadataNode = record.createElement("Metadata");
```

Comment:
I think this element should have the schemaUri from the metadataContainer in the fragment eg. `schemaUri="http://id.kb.dk/schemas/supplementary_tvmeter_metadata"`

Reply:
all attributes from the metadataContainer element in the fragment is copied to the Metadata element (see lines 75-80)